docs/userguide/networking/get-started-overlay.md

```diff
@@ -0,0 +1,247 @@
+<!--[metadata]>
++++
+title = "Get started with multi-host networking"
+description = "Use overlay for multi-host networking"
+keywords = ["Examples, Usage, network, docker, documentation, user guide, multihost, cluster"]
+[menu.main]
+parent = "smn_networking"
+weight=-3
++++
+<![end-metadata]-->     
+
+# Get started with multi-host networking
+
+This article uses an example to explain the basics of creating a mult-host
+network. Docker Engine supports this out-of-the-box through the `overlay`
+network.  Unlike `bridge` networks overlay networks require some pre-existing
+conditions before you can create one. These conditions are:
+
+* Access to a key-value store. Engine supports Consul, Etcd, Zookeeper (Distributed store), and BoltDB (Local store) key-value stores.
+* A cluster of hosts with connectivity to the key-value store.
+* A properly configured Engine `daemon` on each host in the cluster.
+
+You'll use Docker Machine to create both the the key-value store server and the host cluster. This example creates a Swarm cluster.
+
+## Prerequisites
+
+Before you begin, make sure you have a system on your network with the latest version of Docker Engine and Docker Machine installed. The example also relies on VirtualBox. If you installed on a Mac or Windows with Docker Toolbox, you have all of these installed already.
+
+If you have not already done so, make sure you upgrade Docker Engine and Docker Machine to the latests versions.
+
+
+## Step 1: Set up a key-value store
+
+An overlay network requires a key-value store. The key-value store information about the network state which includes discovery, networks, endpoints, ip-addresses, and more. Engine supports Consul, Etcd, and Zookeeper (Distributed store) key-value stores. This example uses Consul.
+
+1. Log into a system prepared with the prerequisite Docker Engine, Docker Machine, and VirtualBox software.
+
+2. Provision a VirtualBox machine called `mh-keystore`.  
+
+				$ docker-machine create -d VirtualBox mh-keystore
+
+		When you provision a new machine, the process adds Docker Engine to the
+		host. This means rather than installing Consul manually, you can create an
+		instance using the [consul image from Docker
+		Hub](https://hub.docker.com/r/progrium/consul/). You'll do this in the next step.
```

Comment:
I agree it's not ideal to recommend a 3rd party image in the docs, but in this case it's probably reasonably OK.  If there were someone interested in maintaining an "official" consul image, it'd definitely be welcome, but it certainly shouldn't be a blocker, and I'd prefer if upstream were at least consulted first if not directly involved in such a thing.